control-announcements/src/main/java/org/oskari/announcements/helpers/AnnouncementsDBHelper.java

```diff
@@ -0,0 +1,367 @@
+        String user = PropertyUtil.get(PROPERTY_DB_USER);
+        String password = PropertyUtil.get(PROPERTY_DB_PASSWORD);
+        Connection connection = DriverManager.getConnection(url, user, password);
+        return connection;
+    }
+
+    /**
+     * Handle getting announcements from database
+     * @param params
+     * @return
+     * @throws JSONException
+     * @throws ActionParamsException
+     */
+    public static JSONObject getAnnouncements() throws JSONException, ActionParamsException {
+        Connection conn = null;
+
+        JSONArray results = new JSONArray();
+        String sql = "";
+        String sqlWithParams = "";
+
+        try  {
+            conn = getConnection();
+            StringBuilder sb = new StringBuilder();
+            sb.append("SELECT id, title, content, begin_date, end_date, active ");
+            sb.append("FROM oskari_announcements ");
+            sb.append("WHERE begin_date <= ? ::DATE AND end_date >= ? ::DATE ");
+            sb.append("ORDER BY id DESC;");
+
+            sql = sb.toString();
+            PreparedStatement pstmt = conn.prepareStatement(sql);
+
+            //Set current date to where clause so we don't get outdated announcements
+            Date date = Date.valueOf(LocalDate.now());
+            pstmt.setDate(1, date);
+            pstmt.setDate(2, date);
+
+            sqlWithParams = pstmt.toString();
+            ResultSet rs = pstmt.executeQuery();
+            SimpleDateFormat df = new SimpleDateFormat("yyyy-MM-dd");
+            while (rs.next()) {
+                JSONObject row = new JSONObject();
+                row.put("id", rs.getInt("id"));
+                row.put("title", rs.getString("title"));
+                row.put("content", rs.getString("content"));
+                row.put("begin_date", df.format(rs.getDate("begin_date")));
+                row.put("end_date", df.format(rs.getDate("end_date")));
+                row.put("active", rs.getBoolean("active"));
+                results.put(row);
+            }
+
+        } catch (SQLException e) {
+            LOG.error(e, "Cannot create SQL query, sql=" + sqlWithParams);
+
+        } finally {
+            if(conn != null) {
+                try {
+                    conn.close();
+                } catch (Exception e) {
+                    LOG.error(e);
+                }
+            }
+        }
+
+        JSONObject json = new JSONObject();
+        json.put("data", results);
+
+        return json;
+
+    }
+
+    /**
+     * Handle getting admin-announcements from database
+     * @param params
+     * @return
+     * @throws JSONException
+     * @throws ActionParamsException
+     */
+    public static JSONObject getAdminAnnouncements() throws JSONException, ActionParamsException {
+        Connection conn = null;
+
+        JSONArray results = new JSONArray();
+        String sql = "";
+        String sqlWithParams = "";
+
+        try  {
+            conn = getConnection();
```

Comment:
You can do try-with-resources and get rid of the finally-block:
```suggestion
        try (Connection conn = getConnection()) {
```